tests/components/smtp/test_notify.py

```diff
@@ -162,5 +162,30 @@ def test_send_text_message(hass, message):
     sample_email = "<mock@mock>"
     message_data = "Test msg"
     with patch("email.utils.make_msgid", return_value=sample_email):
-        result = message.send_message(message_data)
+        result, _ = message.send_message(message_data)
         assert re.search(expected, result)
+
+
+@pytest.mark.parametrize(
+    "target",
+    [
+        None,
+        "target@example.com",
+    ],
+    ids=[
+        "Verify we can send email to default recipient.",
+        "Verify email recipient can be overwritten by target arg.",
+    ],
+)
+def test_send_target_message(target, hass, message):
+    """Verify if we can send email to correct recipient."""
+    sample_email = "<mock@mock>"
+    message_data = "Test msg"
+    with patch("email.utils.make_msgid", return_value=sample_email):
+        if not target:
+            expected_recipient = ["recip1@example.com", "testrecip@test.com"]
+        else:
+            expected_recipient = target
+
+        _, recipient = message.send_message(message_data, target=target)
```

Comment:
Please set up the integration and patch the library. Then call the notify service and assert that the mock library is called as expected. We don't want to interact with integration details directly in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations